neotest_go/three_level_nested_test.go

```diff
@@ -0,0 +1,20 @@
+package main
+
+import "testing"
+
+func TestOdd(t *testing.T) {
+	t.Run("odd", func(t *testing.T) {
+		t.Run("5 is odd", func(t *testing.T) {
+			if 5%2 != 1 {
+				t.Error("5 is actually odd")
+			}
+		})
+		t.Run("7 is odd", func(t *testing.T) {
+			if 7%2 != 1 {
+				t.Error("7 is actually odd")
+			}
+		})
+
+	})
+
+}
```

Comment:
@sergii4 You did not use a three-level test, it's just two levels deep. You should try this:

```go
package main

import "testing"

func TestOdd(t *testing.T) {
	t.Run("odd", func(t *testing.T) {
		t.Run("5 is odd", func(t *testing.T) {
			if 5%2 != 1 {
				t.Error("5 is actually odd")
			}
			t.Run("7 is odd", func(t *testing.T) {
				if 7%2 != 1 {
					t.Error("7 is actually odd")
				}
			})
		})
	})
}
```
<img width="1155" alt="Screenshot 2024-02-24 at 14 34 33" src="https://github.com/nvim-neotest/neotest-go/assets/994357/6ec54946-ebb9-4699-89b3-472662100124">

Reply:
@sergii4 do you think this is related?
- Issue: https://github.com/nvim-neotest/neotest-go/issues/83
- PR: https://github.com/nvim-neotest/neotest-go/pull/81

I'm thinking that you're getting a variable amount of temporary files written right now.

Reply:
I just checked out @encero's branch in #81 but I still get the JSON output on third `t.Run` level.

Reply:
It seems as neotest/tree-sitter upstream problem to me. I keep looking